### PR TITLE
[WIP] Refactor Icon

### PR DESCRIPTION
**Pull request by yesmeck**
ref #12011

### CODE REVIEW

**Comment by afc163** [comment]
- [ ] We need a migrate tool
- [ ] Add migration instruction to 3.0.0 upgrade notes.

File: components/input-number/index.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import * as React from 'react';
 import classNames from 'classnames';
 import RcInputNumber from 'rc-input-number';
-import Icon from '../icon';
 import { Omit } from '../_util/type';
+import UpOutlined from '../icon/icons/UpOutlined';
```
Comment (afc163) [inline components/input-number/index.tsx]: outlined 是默认风格，XxxOutlined 都直接改为 Xxx 吧。

**Comment by PengJiyuan** [comment]
any progress？

**Comment by nerdyadventurer** [comment]
@yesmeck can we get this done?

**Comment by Floriferous** [comment]
A ton of developers are serving an extra 500kb to all their customers. Would be fantastic to get this moved up the priority ladder! :) 

Sending motivational waves your way :innocent:

**Comment by ztplz** [comment]
I agree that this is a high priority, but we are in a hurry at the end of the year.

**Comment by AustinGreen** [comment]
@afc163 @yesmeck: do you think this will be out in the next release? Thank you for all your hard work

**Comment by martenwallewein** [comment]
Hi, how is the progress here?

**Comment by jinliming2** [comment]
It’s the Spring Festival in China, the Chinese are going through the holidays, maybe it means there won’t be progress in the short term until February 11th.

**Comment by juniorZed** [comment]
Still waiting :+1:

**Comment by Kombustor** [comment]
Any progress? This is one of the biggest issues for using antd in production..

**Comment by zaguiini** [comment]
Merge it...

**Comment by nerdyadventurer** [comment]
@yesmeck @afc163 It's been more than three months, IMHO this is a show stopper for using Ant-Design. Can we get this done soon?

**Comment by pkrefta** [comment]
Hi guys,

Any progress on this ? Can we help some way ?

**Comment by mrc535s** [comment]
is this still under review/development?

**Comment by s123121** [comment]
Guys, this is enormous change. Please be patient! You can click the subscribe button to receive notification when this is merge

**Comment by wendzhue** [comment]
It seems that the new API makes icons tree-shakable at the price of breaking changes. 🌝

**Comment by zaguiini** [comment]
So this was closed? @zombieJ

**Comment by wendzhue** [comment]
@zaguiini The refactoring has been implemented in the dependency repo [@ant-design/icons-angular](https://github.com/ant-design/ant-design-icons).

**Pull request closed without merging**